common/config.gni

```diff
@@ -57,6 +57,7 @@ if (is_ios || is_mac) {
   flutter_cflags_objc = [
     "-Werror=overriding-method-mismatch",
     "-Werror=undeclared-selector",
+    "-fapplication-extension",
```

Comment:
This would be set by a buildroot flag so the engine can be built with or without app extension safety. I put it here for testing.